src/optimizer/plugins/CombineCodingAndQuantityValuesOptimizer.ts

```diff
@@ -53,6 +53,15 @@ export default {
       const rules: fshrules.Rule[] = def.rules; // <-- this assignment makes TypeScript happier in the next chunk of code
       const typeCache: Map<string, fhirtypes.ElementDefinitionType[]> = new Map();
 
+      const ruleMap: { [path: string]: { [caretPath: string]: ExportableCaretValueRule[] } } = {};
+      rules
+        .filter(r => r instanceof ExportableCaretValueRule)
+        .forEach((r: ExportableCaretValueRule) => {
+          ruleMap[r.path] = ruleMap[r.path] || {};
+          ruleMap[r.path][r.caretPath] = ruleMap[r.path][r.caretPath] || [];
+          ruleMap[r.path][r.caretPath].push(r);
```

Comment:
Actually... is there a reason you are storing this as an array? I see below that you only ever access `[0]` -- and because of the way GoFSH works, I don't think you'd ever have any more than one anyway. So couldn't you replace 
```ts
ruleMap[r.path][r.caretPath] = ruleMap[r.path][r.caretPath] || [];
ruleMap[r.path][r.caretPath].push(r);
```
with the more simple (and slightly more efficient):
```ts
ruleMap[r.path][r.caretPath] = r;
```